### ISSUE TITLE
Variable usage not recognized if used inside a named slot and assigned inside an expression passed as attribute

### ISSUE THREAD

**Issue by CherryDT** [sveltejs/eslint-plugin-svelte3#100]
**Shows `'abc' is assigned a value but never used`:**

```html
<script>
  import Test from './Test.svelte'

  let abc = 123
</script>

<Test on:stuff={() => {
  abc = 456
}}>
  <span slot="xyz">
    {abc}
  </span>
</Test>
```

**No error:** (assigned outside of attribute)

```html
<script>
  import Test from './Test.svelte'

  let abc = 123

  function test () {
    abc = 456
  }
</script>

<Test on:stuff={test}>
  <span slot="xyz">
    {abc}
  </span>
</Test>
```

**No error:** (usage outside of slot)

```html
<script>
  import Test from './Test.svelte'

  let abc = 123
</script>

<Test on:stuff={() => {
  abc = 456
}}>
  {abc}
  <span slot="xyz"></span>
</Test>
```

**Comment by CherryDT** [comment]
This also happens after refactoring to the new Svelte 3.35.0 way with `<svelte:fragment>` by the way, so it's not the duplicate slots that caused this (however the weird muted color is nowhere to be seen anymore afterwards):

```html
<script>
  export let topView
  export let closeSelf
  export let closeHigher
  export let openAndReplace

  export let id
  export let title
  export let onSave

  import ResourceView from '../components/ResourceView.svelte'
  import { apiCall } from '../lib/api'
  import ViewCustomerForm from './ViewCustomerForm.svelte'
  import ViewPropertyDetails from './ViewPropertyDetails.svelte'
  import ViewPropertyForm from './ViewPropertyForm.svelte'
  import Detail from '../components/Detail.svelte'

  let username
</script>

<ResourceView
  {topView} {closeSelf} {closeHigher} {openAndReplace}
  {id} {title}
  defaultTitle="Kundendetails"
  {onSave}
  getTitle={record => `${record.company} (#${record.customerNo})`}
  getChildTitle={record => record.name}
  order="name"
  select="name,address,interval"
  filter={JSON.stringify({ customer: id })}
  icon="building"
  handleLoad={async (id, mainRecord, items, keepViews) => {
    const [user] = await apiCall('GET', '/users', { filter: JSON.stringify({ customer: id }), select: 'username' })
    username = user?.username
  }}
  resourcePath="/customers"
  resourceName="Kunde"
  childResourcePath="/properties"
  childResourceName="Liegenschaft"
  childResourceGender="f"
  ViewMainForm={ViewCustomerForm}
  ViewChildDetails={ViewPropertyDetails}
  ViewChildForm={ViewPropertyForm}
  search={(searchQuery, searchRegex, record) => record.name?.match(searchRegex) || record.address?.match(searchRegex)}
  allowEdit={true}
  allowDelete={true}
  canDelete={(mainRecord, items) => !items.length}
  let:record
>
  <svelte:fragment slot="header" let:mainRecord>
    <div class="panel-block">
      <Detail title="Anschrift" icon="envelope" ready={mainRecord?.address}>
        {mainRecord?.address?.split('\n').join('; ')}
      </Detail>
    </div>

    {#if username}
      <div class="panel-block">
        <Detail title="Benutzername" icon="user">
          {username}
        </Detail>
      </div>
    {/if}

    {#if mainRecord?.notes}
      <div class="panel-block">
        <Detail title="Notizen" icon="sticky-note">
          {mainRecord.notes.split('\n').join('; ')}
        </Detail>
      </div>
    {/if}
  </svelte:fragment>

  <span class="list-entry" class:is-disabled={!record.interval}>{record.name}<br /><small>{record.address.split('\n').join('; ')}</small></span>
</ResourceView>
```

**Comment by Conduitry** [comment]
This is very likely caused by the same Svelte bug as #94 is. The compiler is returning the wrong AST.

**Comment by CherryDT** [comment]
I just wonder why it works when moving the assigment outside of the attribute expression then...?

**Comment by Conduitry** [comment]
The whole slotted element is missing from the AST, so anything happening in there isn't seen by ESLint.

**Comment by CherryDT** [comment]
But if that was the case then the second example where it works outside of the slot (but still inside the slotted element) wouldn't make sense, no? I'm confused now

**Comment by Conduitry** [comment]
This should be fixed now in Svelte 3.36.0.
